File: Team04_API/Team04_API/Models/Feedback/Client_Feedback.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Team04_API.Models.Chatbot;
using Team04_API.Models.Ticket;
using Team04_API.Models.Users;

namespace Team04_API.Models.Feedback
{
    public class Client_Feedback
    {
        [Key]
        public int Client_Feedback_ID { get; set; }

        [Required]
        public Guid Client_ID { get; set; }

        [Required, MinLength(25), MaxLength(255)]
        public string? Client_Feedback_Detail { get; set; }

        [ForeignKey(nameof(Chatbot_Log.Chatbot_Log_ID))]
        public int? Chatbot_Log_ID { get; set; } 

        [Required]
        public DateTime Feedback_Date_Created { get; set; }

        public int? Ticket_ID { get; set; }

        // VIRTUAL ITEMS
        public virtual Chatbot_Log? Chatbot_Log { get; set; }
        public virtual Ticket.Ticket? Ticket { get; set; }
        public virtual User User { get; set; }
    }
}

File: Team04_Angular/src/app/Classes/client-feedback.ts
export class ClientFeedback {
    client_feedback_ID?: number;
    client_ID: string;
    client_Feedback_Detail: string;
    chatbot_Log_ID: number;
    feedback_Date_Created!: Date; 
    ticket_ID: number;

    constructor(
        client_ID: string,
        client_Feedback_Detail: string,
        chatbot_Log_ID: number,
        feedback_Date_Created: Date,
        ticket_ID: number
    ) {
        this.client_ID = client_ID;
        this.client_Feedback_Detail = client_Feedback_Detail;
        this.chatbot_Log_ID = chatbot_Log_ID;
        this.feedback_Date_Created = feedback_Date_Created;
        this.ticket_ID = ticket_ID;
    }
}
